### ISSUE TITLE
spring.batch.table-prefix is not used if spring-boot-starter-data-jpa is not on the classpath

### ISSUE THREAD

**Issue by DiabolusExMachina** [spring-projects/spring-boot#10757]
Hello, 

I think I found a Bug in the autoconfiguration of  spring-boot-starter-batch.  

The property **spring.batch.table-prefix** is ignored if the dependency **spring-boot-starter-data-jpa** is not on the classpath. 

We do not use JPA in our company because we have replaced it with jOOQ. So we removed all JPA dependencies from our batch projects. But now we cannot start jobs anymore because we use a custom prefix for the batch tables. 

Every time we start a job we get this error 

> org.springframework.jdbc.BadSqlGrammarException: PreparedStatementCallback; bad SQL grammar [SELECT JOB_INSTANCE_ID, JOB_NAME from BATCH_JOB_INSTANCE where JOB_NAME = ? and JOB_KEY = ?]; nested exception is java.sql.SQLSyntaxErrorException: ORA-00942: Tabelle oder View nicht vorhanden" 
(Table or View not available)

If we add the "spring-boot-starter-data-jpa" depdency the spring.batch.table-prefix is used and everything works as expected. 

I've created a sample project to show the problem
[batch-test.zip](https://github.com/spring-projects/spring-boot/files/1410137/batch-test.zip)

**Comment by wilkinsona** [comment]
Thanks for the sample. This does indeed look like a bug as the auto-configuration of `BasicBatchConfigurer` only occurs when `EntityManagerFactory` is on the classpath. This means that you can work around the problem by adding a dependency on the JPA API, for example:

```xml
<dependency>
    <groupId>org.hibernate.javax.persistence</groupId>
    <artifactId>hibernate-jpa-2.1-api</artifactId>
    <version>1.0.0.Final</version>
</dependency>
```

**Comment by wilkinsona** [comment]
Duplicate of #8877.

@snicoll has already fixed this in 2.0. Given the low impact of the [workaround described above](https://github.com/spring-projects/spring-boot/issues/10757#issuecomment-338938200), I think we should leave this as-is in 1.5.x.
